include/dsn/tool-api/network.h

```diff
@@ -267,13 +267,15 @@ class rpc_session : public ref_counter
     enum session_state
     {
         SS_CONNECTING,
-        SS_NEGOTIATING,
         SS_CONNECTED,
         SS_DISCONNECTED
     };
     ::dsn::utils::ex_lock_nr _lock; // [
     volatile session_state _connect_state;
 
+    bool negotiation_succeed = false;
```

Comment:
Is it need to leave it in the big lock, how about use atomic? Lock too many variables should be very careful to avoid dead lock problems.

Reply:
Yes, if we only add `negotiation_succeed ` here, there is no need to lock. But I will add a resend message queue, which is the messages sent in the duration of negotiation.  I should to ensure the access of `negotiatioin_succeed` and the resend message queue to be atomic.